cloud-controller-manager/do/loadbalancers.go

```diff
@@ -284,7 +284,7 @@ func getCertificateIDFromLB(lb *godo.LoadBalancer) string {
 // annotation on the Service gets newly-updated certificate ID from the
 // Load Balancer.
 func (l *loadBalancers) recordUpdatedLetsEncryptCert(ctx context.Context, service *v1.Service, lbCertID, serviceCertID string) error {
-	if lbCertID != "" && lbCertID != serviceCertID {
+	if lbCertID != "" && serviceCertID != "" && lbCertID != serviceCertID {
```

Comment:
If both cert name and cert ID annotation need to be missing for the cert to get removed, I think the PR needs a tiny tweak, so that
- if both cert ID annotation and cert name annotation are empty, we leave those annotations empty and bail out from  recordUpdatedLetsEncryptCert
- if only cert ID annotation is empty, but cert name is set, then we update cert ID annotation to match the actual LB cert ID

Also, a code comment about 200 responses with empty cert ID would be helpful

more context in https://digitalocean.slack.com/archives/C4HREK3T6/p1756914553227899?thread_ts=1751279243.181349&cid=C4HREK3T6